server/src/main/java/org/opensearch/search/startree/filter/provider/StarTreeFilterProvider.java

```diff
@@ -133,27 +134,31 @@ public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, C
             MappedFieldType mappedFieldType = context.mapperService().fieldType(field);
             DimensionFilterMapper dimensionFilterMapper = mappedFieldType == null
                 ? null
-                : DimensionFilterMapper.Factory.fromMappedFieldType(mappedFieldType);
+                : DimensionFilterMapper.Factory.fromMappedFieldType(mappedFieldType, context);
             if (matchedDimension == null || mappedFieldType == null || dimensionFilterMapper == null) {
                 return null;
-            } else {
-                return new StarTreeFilter(
+            }
+            DimensionFilter dimensionFilter = dimensionFilterMapper.getRangeMatchFilter(mappedFieldType, rangeQueryBuilder);
+            ;
+            // if (matchedDimension instanceof DateDimension) {
```

Comment:
Needs to be removed ?

Reply:
Crept in accidentally.
I'll resolve this thread after putting it back.